app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,24 @@
+class SessionsController < ApplicationController
+
+	def new; end
+	
+	def create	
+		auth_hash = request.env['omniauth.auth']
+		if auth_hash["uid"]
+      @user = User.find_or_create_from_omniauth(auth_hash)
+      if @user
+        session[:user_id] = @user.id
+        redirect_to root_path
+      else
+        redirect_to root_path, notice: "Failed to save the user"
+      end
```

Comment:
I don't think I've ever used `notice` before. I just looked it up. Cool! 

I might make the message more descriptive for your user. Why did it fail? What can they do about it?